enumerable_methods.rb

```diff
@@ -0,0 +1,156 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    for i in self
+      yield i
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    ind = 0
+    for i in self
+      yield(i, ind)
+      ind += 1
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_arr = []
+    my_each do |n|
+      new_arr << n if yield(n)
+    end
+    new_arr
+  end
+
+  def my_all?
+    # return true if empty array is passed, regardless of block_given
+    return true if (self.class == Array && count.zero?) || (!count.zero? && !block_given?)
+
+    my_each do |n|
+      return false unless yield(n)
+    end
+    true
+  end
```

Comment:
* [ ] should have working `my_all?`  identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_all? 

- [ ] when no block or argument is given returns true only when none of the collection members are false or nil, if the mentioned condition is not met it, should return false

```ruby     
[nil, false, true, []].my_all? == [nil, false, true, []].all? => false
```
- Yours is giving the wrong answer
![image](https://user-images.githubusercontent.com/52209067/80322251-1ea8ed00-87e1-11ea-8c20-b305b785abf6.png)

- Real all?
![image](https://user-images.githubusercontent.com/52209067/77688266-67148700-6f65-11ea-82e5-cbef16444817.png)

- [ ] when a class is passed as an argument returns true only if all of the collection is a member of such class, otherwise, it should return false

```ruby     
[1, 2.5, 'a', 9].my_all?(Integer) == [1, 2.5, 'a', 9].all?(Integer) => false
```
- Yours is retrieving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/80322264-3e401580-87e1-11ea-8c08-6989ab835bc8.png)

- Real all?
![image](https://user-images.githubusercontent.com/52209067/77688467-be1a5c00-6f65-11ea-8970-15c9e4f4913f.png)

- [ ] when a Regex is passed as an argument returns true if all of the collection matches the Regex

```ruby     
%w[dog door rod blade].my_all?(/d/) == %w[dog door rod blade].all?(/d/) => true
```
- Yours is retrieving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/80322318-79424900-87e1-11ea-88ad-3a1d85d1bfc0.png)

- Real all?
![image](https://user-images.githubusercontent.com/52209067/77688741-31bc6900-6f66-11ea-90ec-b6a7cee7e6ca.png)

- [ ] when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern

```ruby     
[3,4,7,11].my_all?(3) == [3,4,7,11].all?(3)  => false 
```
- Yours is retrieving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/80322338-9bd46200-87e1-11ea-9131-87e96f28a9d1.png)

- Real all?
![image](https://user-images.githubusercontent.com/52209067/77689001-924ba600-6f66-11ea-9b72-210b3bb1d78b.png)